### PR TITLE
Fix up and clean Corne Keyboard code

### PR DESCRIPTION
**Pull request by drashna**
And additional fixing up.

Specifically, this fixes a few things: 
* Using `SRC` for serial (and i2c) causes issues with communication when LTO is enabled. `QUANTUM_LIB_SRC` bypasses the optimization, and this issue.
* Moves includes into the `keyboard.h` file for cleaner keymaps. 
* Cleans up some legacy split code (name audio)

All in all, it makes it cleaner, and fixes some bugs. 

Caveat: My keymap is too large. Travis CI will error out on it. 

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/default/config.h [hunk 1]
```diff
@@ -35,10 +35,12 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#define RGBLIGHT_SAT_STEP 17
-#define RGBLIGHT_VAL_STEP 17
+#ifdef RGBLIGHT_ENABLE
+#    undef RGBLED_NUM
+#    define RGBLIGHT_ANIMATIONS
+#    define RGBLED_NUM 27
+#    define RGBLIGHT_LIMIT_VAL 120
+#    define RGBLIGHT_HUE_STEP 10
+#    define RGBLIGHT_SAT_STEP 17
+#    define RGBLIGHT_VAL_STEP 17
```
Comment (vomindoraan) [inline keyboards/crkbd/keymaps/default/config.h]: I don't recommend separating preprocessor directives from the `#`s. Instead, indent the whole directive, along with the `#`.

**Comment by zvecr** [comment]
Can we clarify which part of the PR fixes the LTO issues?

File: keyboards/crkbd/keymaps/default/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/crkbd/keymaps/default/config.h]: blame clang..... :(

**Comment by drashna** [comment]
That better?

File: keyboards/crkbd/keymaps/default/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/crkbd/keymaps/default/config.h]: Also, is there a specific reason for not doing so?

Comment (zvecr) [reply keyboards/crkbd/keymaps/default/config.h]: In my own experience, it makes searching for `#define` difficult. The current coding conventions mention `Do not put whitespace between # and if` which I think should apply to defines too.

Comment (vomindoraan) [reply keyboards/crkbd/keymaps/default/config.h]: Agreed. If `clang-format` goes against this, it should be changed.

Comment (drashna) [reply keyboards/crkbd/keymaps/default/config.h]: Will fix once I am home

Comment (drashna) [reply keyboards/crkbd/keymaps/default/config.h]: Fixed.

**Comment by drashna** [comment]
@zvecr that better?

File: keyboards/crkbd/rev1/rev1.h [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-#ifdef RGBLIGHT_ENABLE
-//rgb led driver
-#include "ws2812.h"
+#ifdef PROTOCOL_LUFA
+  #include "lufa.h"
+  #include "split_util.h"
+#endif
+
+#ifdef SSD1306OLED
+  #include "ssd1306.h"
```
Comment (vomindoraan) [inline keyboards/crkbd/rev1/rev1.h]: This should probably use 4 space indentation. Same goes for the rest of the file (lines 9–10, 27–41, 44–54).

```suggestion
    #include "ssd1306.h"
```

Comment (drashna) [reply keyboards/crkbd/rev1/rev1.h]: Fixed.  That better?

Comment (vomindoraan) [reply keyboards/crkbd/rev1/rev1.h]: Yep, that's better. I see you opted to keep the 2-space indentation in the layout macros, why is that?

Comment (drashna) [reply keyboards/crkbd/rev1/rev1.h]: Because I switched to macOS, and for whatever reason, it wants to ignore clang format setting, apparently....

**Comment by snowe2010** [comment]
yay!

**Pull request merged**